borsh/src/ser/helpers.rs

```diff
@@ -0,0 +1,23 @@
+use crate::maybestd::{
+    io::{Result, Write},
+    vec::Vec,
+};
+use crate::BorshSerialize;
+
+/// Serialize an object into a vector of bytes.
+#[inline]
+pub fn serialize_to_vec<T>(value: &T) -> Result<Vec<u8>>
+where
+    T: BorshSerialize + ?Sized,
+{
+    value.try_to_vec()
+}
+
+/// Serializes an object directly into a `Writer`.
+#[inline]
+pub fn serialize_to_writer<T, W: Write>(value: &T, mut writer: W) -> Result<()>
```

Comment:
Let's add an integration test for these, just to make sure that we accidently don't remove the re-export in minor release. I  don't see a good existing test for this, I suggest creating `borsh/tests/smoke.rs`.

Reply:
added smoke tests